packages/sharing-smart-contract/test/e2e/consumeProtectedData.test.js

```diff
@@ -137,47 +137,138 @@ describe('ConsumeProtectedData', () => {
-          .subscribeToCollection(collectionTokenId, subscriptionParams);
+          .consumeProtectedData(protectedDataAddress, workerpoolOrder, appAddress, true),
+      ).to.be.reverted;
+    });
+  });
+
+  describe('voucher - consumeProtectedData()', () => {
+    it('should create a deal on chain and consume asset in the user voucher', async () => {
+      const {
+        dataProtectorSharingContract,
+        pocoContract,
+        protectedDataAddress,
+        appAddress,
+        collectionTokenId,
+        subscriptionParams,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({ dataProtectorSharingAddress });
+
+      await pocoContract
+        .connect(voucherOwner)
+        .approve(await dataProtectorSharingContract.getAddress(), subscriptionParams.price);
+      await pocoContract.connect(voucherOwner).deposit({
+        value: ethers.parseUnits(subscriptionParams.price.toString(), 'gwei'),
+      }); // value sent should be in wei
+
+      await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .subscribeToCollection(collectionTokenId, subscriptionParams);
+
+      const tx = await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .consumeProtectedData(protectedDataAddress, workerpoolOrder, appAddress, true);
+      await tx.wait();
+
+      expect(tx)
+        .to.emit(dataProtectorSharingContract, 'ProtectedDataConsumed')
+        .withArgs((_dealId, _protectedDataAddress, _mode) => {
+          assert.equal(_dealId.constructor, ethers.Bytes32, 'DealId should be of type bytes32');
+          assert.equal(_protectedDataAddress, protectedDataAddress, 'DealId should be of type bytes32');
+          assert.equal(_mode, 0, 'Mode should be SUBSCRIPTION (0)');
+        });
+    });
+
+    it('should revert if the consumer has approved contract but account balance is insufficient', async () => {
+      const {
+        dataProtectorSharingContract,
+        pocoContract,
+        protectedDataAddress,
+        appAddress,
+        collectionTokenId,
+        subscriptionParams,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({
+        dataProtectorSharingAddress,
+        workerpoolprice: 1,
+      });
 
-        const tx = await dataProtectorSharingContract
+      await pocoContract
+        .connect(voucherOwner)
+        .approve(await dataProtectorSharingContract.getAddress(), subscriptionParams.price);
+      await pocoContract.connect(voucherOwner).deposit({
+        value: ethers.parseUnits(subscriptionParams.price.toString(), 'gwei'),
+      }); // value sent should be in wei
```

Comment:
as the deal is supposed to be paid by the voucher, the user doesn't need to deposit on its account